include/model.php

```diff
@@ -150,10 +150,12 @@ public function get_languages_list( $args = array() ) {
 			 * /!\ This filter is fired *before* the $polylang object is available.
 			 *
 			 * @since 1.8
+			 * @since 3.4 Deprecated.
+			 * @deprecated
 			 *
 			 * @param PLL_Language[] $languages The list of language objects.
 			 */
-			$languages = apply_filters( 'pll_after_languages_cache', $languages );
+			$languages = apply_filters_deprecated( 'pll_after_languages_cache', array( $languages ), '3.4', 'pll_additional_language_data' );
```

Comment:
Could we add information with the `$message` parameter?
If I understood correctly, this filter was created to avoid getting cached values (for URLs at least).
I'm not sure `pll_additional_language_data` is an ideal replacement here. Maybe we could steer users to use the proper getters?

Reply:
But which hook would be more suited? (it's the one you use in https://github.com/polylang/polylang-wc/pull/548 😏😜).

Reply:
I know 😅 I'm a bit confused too. Let's say I know what I'm doing (really?!?). But third party might want to use `pll_additional_language_data` instead of `pll_after_languages_cache` thinking it would allow them to bypass cached properties.
Also, I purely removed `pll_after_languages_cache` in PLLWC in favor of the getters.

Reply:
Currently there is no way to bypass the cache, it would require filters in `get_home_url()` and `get_search_url()`, wouldn't it?

Reply:
I think that's already the case [here](https://github.com/polylang/polylang/blob/3.4-dev/include/language.php#L695-L702), right?

Reply:
Well, I meant a dedicated PLL filter but filtering `site_url()` is enough indeed.
So you think we should add a message about the filter `'site_url'`?

Reply:
After discussing we decided to remove the replacement for the hook `'pll_after_languages_cache'` and add a comment about using `'site_url'` instead.
Done in [9d7554b](https://github.com/polylang/polylang/pull/1210/commits/9d7554bc6e961d14b5a8b92c1556bd3ed8053e7c).